### PR TITLE
add image for cluster-api-actuator-pkg to run regression testing

### PR DESCRIPTION
**Pull request by sunzhaohua2**
Build an image so that we can use `make test-e2e` in step to run regression tests on qe jobs
@JoelSpeed @shellyyang1989 @miyadav @huali9 PTAL

### CODE REVIEW

File: ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+base_images:
+  openshift_release_golang-1.21:
+    name: release
+    namespace: openshift
+    tag: golang-1.21
+build_root:
+  from_repository: true
+images:
+- dockerfile_path: Dockerfile.ci
```
Comment (shellyyang1989) [inline ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:9]: Thanks Zhaohua for initiating the effort. Would you create a separate qe customized Dockerfile later?

Comment (JoelSpeed) [reply ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:9]: Do we have a need to create a customised image? The current one copies the entire source in, so, we will have access to both QE and dev tests from there

Comment (JoelSpeed) [reply ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:9]: This image is only used for testing, it doesn't ship as part of the product so I think I'm ok to keep them together even if we need to add new tools

Comment (shellyyang1989) [reply ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:9]: ah, that would be great! Thank you.

File: ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  inputs:
+    openshift_release_golang-1.21:
+      as:
+      - registry.ci.openshift.org/openshift/release:golang-1.21
+  to: cluster-api-actuator-pkg-test
```
Comment (damdo) [inline ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:14]: Should we use 1.22 https://github.com/openshift/cluster-api-actuator-pkg/blob/086923b9d3fd72a50a6c1587f8dc1189cae48df4/go.mod#L3 ?

File: ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+base_images:
+  openshift_release_golang-1.21:
+    name: release
+    namespace: openshift
+    tag: golang-1.21
```
Comment (damdo) [inline ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:5]: Should we use 1.22 https://github.com/openshift/cluster-api-actuator-pkg/blob/086923b9d3fd72a50a6c1587f8dc1189cae48df4/go.mod#L3 ?

Comment (sunzhaohua2) [reply ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:5]: Thanks @damdo we should update https://github.com/openshift/cluster-api-actuator-pkg/blob/master/Dockerfile.ci as well?

Comment (damdo) [reply ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml:5]: Yeah sounds reasonable. Thank you!

File: ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml (hunk 2 above)
Comment (sunzhaohua2) [reply ci-operator/config/openshift/cluster-api-actuator-pkg/openshift-cluster-api-actuator-pkg-release-4.18__images.yaml]: Updated, thank you.

**Comment by damdo** [comment]
Does this need to be added also to the master pre/post submits?

**Comment by sunzhaohua2** [comment]
> Does this need to be added also to the master pre/post submits?

Thank you @damdo I updated, please help to take a look again.

**Comment by shellyyang1989** [comment]
> Does this need to be added also to the master pre/post submits?

Good question. IMHO, the image is intended for periodic QE CI jobs on supported and pre-release 4.y versions, so it should be included for the 4.y versions. If the change of master branch is synced to release-4.18 branch automatically, then seems like it's not needed to add it to master. Please correct me if I'm wrong.

**Comment by damdo** [comment]
Ok sounds reasonable.

/lgtm

/assign @JoelSpeed 

For approval

**Comment by damdo** [comment]
@sunzhaohua2 you might want to add rehearsal ack once you are happy

**Pull request merged**